crates/flipperzero/src/dialogs/mod.rs

```diff
@@ -45,12 +52,34 @@ impl DialogsApp {
     }
 
     /// Displays a message.
-    pub fn show(&mut self, message: &DialogMessage) -> DialogMessageButton {
+    pub fn show_message(&mut self, message: &DialogMessage) -> DialogMessageButton {
         let button_sys =
             unsafe { sys::dialog_message_show(self.data.as_ptr(), message.data.as_ptr()) };
 
         DialogMessageButton::from_sys(button_sys).expect("Invalid button")
     }
+
+    /// Displays a file browser.
+    ///  - path is a optional preselected file path
+    ///  - options are optional file browser options
+    pub fn show_file_browser(
+        &mut self,
+        path: Option<&FuriString>,
+        options: Option<&DialogFileBrowserOptions>,
+    ) -> Option<FuriString> {
+        let result_path = FuriString::new();
+        // path will be unmodified but needs to be a valid FuriString.
+        // We can reuse the empty result_path if path is not provided.
+        let path = path.unwrap_or(&result_path).as_ptr();
+        let options = match options {
+            Some(opts) => &opts.data as *const sys::DialogsFileBrowserOptions,
+            None => ptr::null(),
+        };
+        unsafe {
+            sys::dialog_file_browser_show(self.data.as_ptr(), result_path.as_ptr(), path, options)
```

Comment:
I fell like this is unsound:

1. `options` is passed by `&` and the return type has no lifetimes;
2. you convert `&opts.data` to a `const`-pointer;
3. you return from the function thus the original `&DialogFileBrowserOptions` can now be moved.

Now if the API user moves the `DialogFileBrowserOptions` to which he pointed in the constructor, it will include moving its `data` field which will invalidate the `options` pointer passed to `dialog_file_browser_show`

Reply:
I think this is ok, but I'm not an expert on Rust lifetimes.
1. `options` is only used as a reference and is unmodified by this Rust function or the `dialog_file_browser_show` C function. The C function is blocking, when it returns, we are done with the `options` and it can be dropped by the caller when it goes out of scope. So I don't think we need to move `options` into `show_file_browser` or hold onto a reference.
2. Yep, `dialog_file_browser_show` wants a const pointer to a `DialogsFileBrowserOptions` (the inner C struct). The const designates that C won't modify this struct and we can inspect the C code to know that it won't hold onto this pointer after the `dialog_file_brower_show` function returns.
3. The caller can do whatever they want to the original `DialogFileBrowserOptions` after this function, no data will be modified from underneath them.

Reply:
Hmm, I haven't thought that `dialog_file_browser_show` blocks until it has done all its work.

In this case, yes, everyhting seems fine :rocket: